src/validator/config_validator.py

```diff
@@ -197,26 +197,57 @@ def validate_pool_sizes_non_zero(shared_args):
     return errors
 
 
-def validate_job_size_non_zero(shared_args):
-    """ Ensure the specified job size is a non-zero numbers.
+def validate_job_size(shared_args, factory_definitions):
+    """ Ensure the specified job size is greater than zero but less than the
+    smallest maximum number of records per file
 
     Parameters
     ----------
     shared_args : dict
         Dictionary of the "shared_config" section of the config file
+    factory_definitions : dict
+        Dictionary of string:dict key/value pairs where keys are names of
+        domain objects, and each value is a dictionary containing the
+        configuration settings for that domain object.
 
     Returns
     -------
     list
         List of a single object, an error message, if job size strictly
-        less than zero, contains nothing otherwise.
+        less than zero or greater than the smallest maximum number of records
+        per file across all domain objects, contains nothing otherwise.
     """
 
     error = []
     job_size = shared_args['number_of_records_per_job']
-    if job_size <= 0:
-        error = ["- Pool job size must be a positive value"]
-    return error
+    try:
+        maximum_job_size = min(
+            [
+                config['max_objects_per_file']
+                for config in factory_definitions.values()
+                if config['fixed_args']['record_count'] > 0
+            ]
+        )
+
+        if not 0 < job_size <= maximum_job_size:
+            error = [
+                "- Pool job size must be greater than 0 but not greater " +
+                "than the smallest 'max_objects_per_file' value in the " +
+                f"config ({maximum_job_size})"
```

Comment:
Make the end of this a bit clearer, maybe something like "The current smallest 'max_objects_per_file' value in the config is x" (maybe even mention the domain object as well).  Otherwise it's a bit ambiguous as the number in brackets could be a line number or an error code